File: csharp/main/DecoratorUsage.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class DecoratorUsage
	{
		public List<DecoratorTarget> ValidOn { get; } = new List<DecoratorTarget>();
		public Boolean Repeatable { get; set; } = false;

		public static DecoratorUsage Default()
		{
			DecoratorUsage usage = new DecoratorUsage();
			foreach (DecoratorTarget target in Enum.GetValues(typeof(DecoratorTarget)))
				usage.ValidOn.Add(target);
			return usage;
		}
	}
}

File: java/src/main/java/deltix/luminary/DecoratorUsage.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class DecoratorUsage {
    private final List<DecoratorTarget> validOn = new ArrayList<>();
    private boolean repeatable = false;

    public List<DecoratorTarget> getValidOn() {
        return validOn;
    }

    public boolean isRepeatable() {
        return repeatable;
    }

    public void setRepeatable(boolean repeatable) {
        this.repeatable = repeatable;
    }

    public static DecoratorUsage byDefault() {
        final DecoratorUsage usage = new DecoratorUsage();
        usage.getValidOn().add(DecoratorTarget.ENUMERATION);
        usage.getValidOn().add(DecoratorTarget.ENUMERATION_MEMBER);
        usage.getValidOn().add(DecoratorTarget.INTERFACE);
        usage.getValidOn().add(DecoratorTarget.INTERFACE_PROPERTY);
        usage.getValidOn().add(DecoratorTarget.CLASS);
        usage.getValidOn().add(DecoratorTarget.CLASS_PROPERTY);
        usage.getValidOn().add(DecoratorTarget.DECORATOR);
        usage.getValidOn().add(DecoratorTarget.DECORATOR_PROPERTY);
        usage.getValidOn().add(DecoratorTarget.INTERFACE_METHOD);
        usage.getValidOn().add(DecoratorTarget.FORMAL_PARAMETER);
        return usage;
    }
}
